docker-proxy-rule-core-jdk21/src/main/java/com/palantir/docker/proxy/ForwardingInetAddressResolver.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.docker.proxy;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.net.spi.InetAddressResolver;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+class ForwardingInetAddressResolver implements InetAddressResolver {
+    private final InetAddressResolver delegate;
+    private final InetAddressResolver fallback;
+    private final Supplier<Boolean> delegateEnabled;
```

Comment:
might consider a `BooleanSupplier` to avoid auto-boxing, not a big deal for test code though